JavaScriptCode.js

```diff
@@ -0,0 +1,368 @@
+var _ = require('lodash');
+const prompt = require('prompt-sync')();
+let str = true;
+let checkordernumber = () => {
+    while (str)
+    {
+        const ordernumber = prompt('Please enter your order : ');
+        if (!isNaN(ordernumber) && ordernumber.toString().indexOf('.') == -1 && ordernumber.toString().indexOf('#') == -1){
+            if(ordernumber >= 1){
+                return parseInt(ordernumber);
+                str = false;
+            }
+            else{
+                console.log(`The order number should be higher than 1.`);
+            }
+        }
+        else{
+            console.log(`Please enter a correct order number where order number should be an integer number higher than 1.`);
+        }
+    }
+}
+
+let myordernumber = checkordernumber();
+console.log(`Hey the order number is : ` + myordernumber);
+const largeboxobject = {size:'Large',value:20,price:1.80};
+const mediumboxobject = {size:'Medium',value:10,price:1.00};
+const smallboxobject = {size:'Small',value:5,price:0.6};
+const bagprice = 5.50;
+const currency = '$';
+let orderId = 1;
+
+/////////////////////////////////////////////////////
+// Start of part A
+/////////////////////////////////////////////////////
+let boxesarrayfun = (par) => {
+    let boxesarray = [];
+    boxesarray.push({size:largeboxobject.size,numbers_of_Bags: largeboxobject.value * parseInt(par / largeboxobject.value), price_of_Boxes:largeboxobject.price * parseInt(par / largeboxobject.value),quantity_of_Size : parseInt(par / largeboxobject.value)});
+    par = par % largeboxobject.value;
+    boxesarray.push({size:mediumboxobject.size,numbers_of_Bags: mediumboxobject.value * parseInt(par / mediumboxobject.value), price_of_Boxes:mediumboxobject.price * parseInt(par / mediumboxobject.value),quantity_of_Size : parseInt(par / mediumboxobject.value)});
+    par = par % mediumboxobject.value;
+    if(par > smallboxobject.value) {
+        boxesarray.push({size:smallboxobject.size,numbers_of_Bags: par, price_of_Boxes:smallboxobject.price * 2,quantity_of_Size : 2});
+    }
+    else if (par > 0){
+        boxesarray.push({size:smallboxobject.size,numbers_of_Bags:  par, price_of_Boxes:smallboxobject.price * 1,quantity_of_Size : 1});
+    }
+    else{
+        boxesarray.push({size:smallboxobject.size,numbers_of_Bags:  par, price_of_Boxes:smallboxobject.price * par,quantity_of_Size : par});
+    }
+    return boxesarray;
+}
+
+let order = (par) => {
+    let bagp = bagprice;
+    let boxesarray = [];
+    let totalnumberofbags, totalcoffeeprice;
+    let totalbagsprice, totalorderprice;
+    let finalobject = {};
```

Comment:
why not declare the object with everything directly?
const finalObject = {
orderID: orderId
etc...
{

Reply:
yes your write much better noted thank u

Reply:
done